src/lib-components/BlockRemoveSearchFilter.vue

```diff
@@ -0,0 +1,121 @@
+<script setup>
+// Helpers
+import { computed } from 'vue'
+import { useRoute } from 'vue-router'
+import SvgGlyphClose from 'ucla-library-design-tokens/assets/svgs/icon-close.svg'
+import getSectionName from '@/utils/getSectionName'
+
+const props = defineProps({
+  title: {
+    type: String,
+    default: '',
+  },
+  filterType: {
+    type: String,
+    default: '',
+  },
+  index: {
+    type: Number,
+    default: 0,
+  },
+  color: {
+    type: String,
+    default: '', // This will be "visit", "about", "help".
+  },
+},)
+
+const emit = defineEmits(['removeBlockFilter'])
+
+const route = useRoute()
+
+const sectionName = computed(() => {
+  return (
+    props.color
+        || (route.path
+          ? getSectionName(route.path)
+          : 'color-default')
+  )
+})
+
+const classes = computed(() => {
+  return ['block-remove-search-filter', `color-${sectionName.value}`]
+})
+
+function closeBlockFilter() {
+  emit('removeBlockFilter', props.index)
+}
+</script>
+
+<template>
+  <span :class="classes">
+    <a>
+      {{ title }}
+    </a>
+    <button type="button" class="button-close" @click="closeBlockFilter">
+      <SvgGlyphClose class="svg-glyph-close" />
+    </button>
+  </span>
+</template>
+
+<style lang="scss" scoped>
+.block-remove-search-filter {
+    padding: 12px;
+    display: flex;
+    flex-direction: row;
+    border: 1.5px var(--color-border) solid;
+    background-color: #ffffff;
+    border-radius: 4px;
+    align-items: center;
+    width: fit-content;
+    font-weight: 400;
+    font-size: 18px;
+    line-height: 100%;
+
+    .button-close {
+        outline: none; // focus-visible was adding a border
+    }
+
+    // background-color variable is not translating the sass which is the reason for the repetition
+
+    &:hover {
+        cursor: pointer;
+
+        &.color-default {
```

Comment:
Rather than repeating selectors hover states can be nested in the parent/target selectors: 

```
&.color-default {
        --color-border: var(--color-default-cyan-03);
        &:hover {
          background-color: transparentize(($default-cyan-03), 0.7);
        }
    }
```
A transparent value of 0.7 or 0.8 is a better match to prod.

Reply:
I like the refactor. I was working with Axa on the transparency color. I'll ask her which she prefers.

```
&.color-default {
        --color-border: var(--color-default-cyan-03);
        &:hover {
          background-color: transparentize(($default-cyan-03), 0.7);
        }
    }
```

Reply:
Axa liked the lighter shade. I refactored the css like you suggested. It looks much nicer.